main.py

```diff
@@ -1,404 +1,3 @@
-        key = c["teacher_username"]
-        if not key: # Skip free periods
-            continue
-        if key in available and not is_teacher_schedule(schedule):
-            available[key]["students"] += 1
-        elif schedule["grade"] in grades:
-            available[key] = copy.copy(c)
-            available[key]["students"] = 1
-    return list(available.values())
-
-# Modifies 'available' in place by removing and returning
-# the desired item
-def pop_current_class(available, schedule, term, period):
-    current_class = get_class_by_period(schedule["classes"][term], period)
-    for c in available:
-        if c["teacher_username"] == current_class["teacher_username"]:
-            available.remove(c)
-            return c
-
-def get_class_by_period(schedule, period):
-    for c in schedule:
-        if c["period"].lower() == period.lower():
-            return c
-
-'''@app.route('/cron/<job>')
-class CronHandler():
-    def get(self, job):  # On url invoke
-        if job == "lunch":
-            update_lunch.read_lunches()
-            self.response.write("Success")
-        elif job == "schedules": # Warning - takes a LONG time
-            json_schedules = fetch_schedules_with_api()'''
-
-# Change and view privacy settings
-@app.route('/privacy', methods=['GET', 'POST'])
-def handle_settings():
-    if 'username' not in session:
-        abort(403)
-    user = get_database_entry(session["username"])
-
-    if request.method == 'GET':
-        return json.dumps(dict(user.items()))
-
-    elif request.method == 'POST':
-        print(request.args.get('share_photo'))
-
-        user.update({
-            "share_photo": request.form['share_photo'] == "true",
-            "share_schedule": request.form['share_schedule'] == "true"
-        })
-        datastore_client.put(user)
-        return json.dumps({})
-
-@app.route('/search/<keyword>')
-def handle_search(keyword):
-    if 'username' not in session:
-        abort(403)
-
-    results = []
-    for schedule in get_schedule_data().values():
-        test_keyword = schedule["firstname"] + " " + schedule["lastname"]
-        if keyword.lower() in test_keyword.lower():
-            results.append({"name": test_keyword, "username": schedule["username"]})
-            if len(results) >= 5:  # We only display five results
-                break
-    return json.dumps(results)
-
-if __name__ == '__main__':
-    # Only used for running locally
-
-    # This is used when running locally only. When deploying to Google App
-    # Engine, a webserver process such as Gunicorn will serve the app. This
-    # can be configured by adding an `entrypoint` to app.yaml.
-    # Flask's development server will automatically serve static files in
-    # the "static" directory. See:
-    # http://flask.pocoo.org/docs/1.0/quickstart/#static-files. Once deployed,
-    # App Engine itself will serve those files as configured in app.yaml.
-    app.run(host='127.0.0.1', port=8080, debug=True)
+init_app()
```

Comment:
This never calls `app.run`, and so won't actually start the server locally.

Reply:
This also won't work on production, because GAE uses a webserver process like Gunicorn - Flask is only for development. If the "main function" to run is not `main()` in `main.py`, you need to configure that in `app.yaml`